### PR TITLE
Fix issue config qos reload causing orchagent aborted via tracking dependencies among QoS tables

### PR DESCRIPTION
**Pull request by stephenxs**
**What I did**
Fix issue `config qos reload` causing orchagent aborted via tracking dependencies among QoS tables

1. Track dependencies among QoS tables.
   Won't call SAI remove API for an object if it is still referenced.
2. Support removing/replacing one field in PORT_QOS_MAP and QUEUE tables.
3. Optimize logic to handle QUEUE table.
4. Remove switch level DSCP_TO_TC map before the map is removed.
5. Add mock test

Signed-off-by: Stephen Sun <stephens@nvidia.com>

**Why I did it**
Fix issue.

**How I verified it**
Manually test and mock test.

**Details if related**

### CODE REVIEW

**Comment by stephenxs** [comment]
This is to fix issue Azure/sonic-buildimage#9088

**Comment by liat-grozovik** [comment]
@neethajohn kindly reminder to review this fix.

**Comment by liat-grozovik** [comment]
@neethajohn kindly reminder to review this fix.

**Comment by dprital** [comment]
@neethajohn - Can you please approve this PR ?

File: tests/mock_tests/qosorch_ut.cpp [hunk 1]
```diff
@@ -0,0 +1,789 @@
...
+        ASSERT_EQ((*QosOrch::getTypeMap()[CFG_PORT_QOS_MAP_TABLE_NAME]).count("Ethernet0"), 0);
+
+        // Make sure the maps can be removed now. Checking anyone should suffice since all the maps are handled in the same way.
+        entries.push_back({"AZURE", "DEL", {}});
+        consumer = dynamic_cast<Consumer *>(gQosOrch->getExecutor(CFG_PFC_PRIORITY_TO_PRIORITY_GROUP_MAP_TABLE_NAME));
+        consumer->addToSync(entries);
+        entries.clear();
+        static_cast<Orch *>(gQosOrch)->doTask();
+        ASSERT_EQ(++current_sai_remove_qos_map_count, sai_remove_qos_map_count);
+        ASSERT_EQ((*QosOrch::getTypeMap()[CFG_PFC_PRIORITY_TO_PRIORITY_GROUP_MAP_TABLE_NAME]).count("AZURE_1"), 0);
```
Comment (neethajohn) [inline tests/mock_tests/qosorch_ut.cpp:749]: should check for "AZURE", not "AZURE_1"

Comment (stephenxs) [reply tests/mock_tests/qosorch_ut.cpp]: Fixed.

**Comment by stephenxs** [comment]
LGTM failed due to
```
[2022-02-21 05:02:32] [build-stdout] [92mProcessing saiversion_8h.xml[0m
[2022-02-21 05:02:32] [build-stdout] [92mProcessing saivirtualrouter_8h.xml[0m
[2022-02-21 05:02:32] [build-stdout] [92mProcessing saivlan_8h.xml[0m
[2022-02-21 05:02:32] [build-stdout] [92mProcessing saiwred_8h.xml[0m
[2022-02-21 05:02:32] [build-stdout] [91mERROR: invalid type '' for SAI_ACL_COUNTER_ATTR_TABLE_ID[0m
[2022-02-21 05:02:32] [build-stdout] [91mERROR: flags are not defined for SAI_ACL_COUNTER_ATTR_TABLE_ID[0m
[2022-02-21 05:02:32] [build-stdout] [91mERROR: FATAL ERROR === MUST FIX === : Can't use an undefined value as an ARRAY reference at parse.pl line 1676.
[2022-02-21 05:02:32] [build-stdout] [0m
[2022-02-21 05:02:32] [build-stderr] make[3]: *** [Makefile:107: saimetadata.c] Error 1
[2022-02-21 05:02:32] [build-stdout] make[3]: Leaving directory '/opt/src/sonic-sairedis/SAI/meta'
[2022-02-21 05:02:32] [build-stderr] make[2]: *** [Makefile:1143: saimeta-gen] Error 2
[2022-02-21 05:02:32] [build-stdout] make[2]: Leaving directory '/opt/src/sonic-sairedis/meta'
[2022-02-21 05:02:32] [build-stderr] make[1]: *** [Makefile:442: install-recursive] Error 1
[2022-02-21 05:02:32] [build-stdout] make[1]: Leaving directory '/opt/src/sonic-sairedis'
[2022-02-21 05:02:32] [build-stderr] dh_auto_install: error: make -j1 install DESTDIR=/opt/src/sonic-sairedis/debian/tmp AM_UPDATE_INFO_DIR=no returned exit code 2
[2022-02-21 05:02:32] [build-stderr] make: *** [debian/rules:59: binary-syncd-vs] Error 25
[2022-02-21 05:02:32] [ERROR] Spawned process exited abnormally (code 2; tried to run: [/opt/work/lgtm-workspace/lgtm/extract.sh])
```
which looks like a common issue.

**Review by neethajohn — APPROVED** [review]

**Pull request merged**